### DISCUSSION TITLE
Geospatial Data Science Internship Queries

### DISCUSSION THREAD

**Discussion by parvathykrishnank** [undpindia/dicra#15] [Q&A] [answered]
Please post your queries here regarding the Geospatial Data Science Internship at UNDP

**Comment by shoebgf0150** [comment]
What is the stipend amount per month?

**Reply by parvathykrishnank** [reply]
Please refer to the ToR in the advertisement for this information.

**Reply by drshaheed** [reply]
> 

hello ma'am the entire work should be in python code or can we use ArcGIS for creating the same?

**Reply by Swetha-Kolluri** [reply]
Entire work should be in python code only.
For the BLOG, you may do data visualization with other tools (if required).

**Comment by Vivek0000007** [comment]
We can do data visualization in any software. Right?

**Reply by parvathykrishnank** [reply]
Please make all the code submissions in Python only (including visualization). But if you want to include visualizations in your blog, feel free to use other software.

**Comment by adiradhu17** [comment]
Respected sir/ma'am
Thank you very much for selecting me for the next round of assessment..!
I have gone through the instructions and all, but it has been asked to submit your assessment before 13th March, Sunday. Unfortunately, currently my mid semester examinations are going on so I wouldn't be able to spare time.
So would it be okay, if I submit on next Sunday i.e. on 20th March or would you provide me another deadline date.
I humbly request you to look into this and do the needful.
Thanking you,
Yours sincerely
Aditya

**Reply by adiradhu17** [reply]
Name - Aditya Dharaiya
Email ID - adiradhu@gmail.com

**Reply by parvathykrishnank** [reply]
In order to be fair to all the applicants, and in accordance with our procurement plan, we cannot give any extension on the deadline.

**Comment by msameeruddin** [comment]
In task 1 we are supposed to get the fire counts as per the districts. In Telangana, there are about 31 districts as per [Wikipedia](https://en.wikipedia.org/wiki/List_of_districts_of_Telangana) and other sources.

![image](https://user-images.githubusercontent.com/63333753/157644671-87c7985c-5c53-4f04-adc0-cbc12b76d7d0.png)

Whereas, if we follow [GADM](https://gadm.org/maps/IND/telangana.html), The number of sub-divisions or districts is less.

![image](https://user-images.githubusercontent.com/63333753/157644971-0c67e8d1-ec28-47cf-960c-7b647c0f92c4.png)

What shall we do?

**Reply by msameeruddin** [reply]
The new shapefile for the state which includes 31 districts is only available as of now.

**Reply by parvathykrishnank** [reply]
Hello, Please use Level 3 in the GADM data as mentioned in the task description.

<img width="955" alt="image" src="https://user-images.githubusercontent.com/42402451/157647535-ee5a2f63-a9db-4524-ad31-2c879dab8534.png">

**Reply by Swetha-Kolluri** [reply]
@YaodongQiu

**Reply by parvathykrishnank** [reply]
@msameeruddin

**Reply by msameeruddin** [reply]
I get your point @parvathykrishnank 

This is how we get the level 3 map of Telangana.

![image](https://user-images.githubusercontent.com/63333753/157856857-139454e0-43cd-4a36-be71-6651508b5e9f.png)

But we do not have adm names to get the JSON output as specified. The result that we get from the level 3 data after reading it using geopandas is below -

![image](https://user-images.githubusercontent.com/63333753/157857415-3ab71199-0ff9-43a6-9b2a-7d4d24cddb6b.png)

There are almost 111 boundaries. In every row, the geometry is different and the rest of the columns are the same. Now, for the output - it is specified that we need to get the adm_name, year, and fireCount. How can we get the adm name if we do not have that column in the first place?

**Reply by maby200** [reply]
Have you made sure you have downloaded the right dataset?

I loaded that file for level3 and it has a column with the following 'NAME_3'. Like this:

![imagen](https://user-images.githubusercontent.com/18635819/157862017-d4971e6e-3efb-4dd1-ae6f-ce8c59083fa0.png)

**Reply by msameeruddin** [reply]
I downloaded the data from this link (https://gadm.org/download_country.html)

> I clicked on the level 3 link (from the above website) and it downloaded the `gadm40_IND_3.kmz` file.

Once it is finished, I loaded it using geopandas.

The file (`gadm40_IND_3.kmz`) has 2336 rows and 12 rows. I had to perform a spatial join with the given geojson file to basically filter only Telangana.

**Reply by msameeruddin** [reply]
@maby200 can you please send a link to get this dataset.

**Reply by maby200** [reply]
That is right, but I suggest you to download the shapefile..

![imagen](https://user-images.githubusercontent.com/18635819/157867100-d214561a-7e90-4ff0-a667-cf9e12b3a837.png)

**Reply by msameeruddin** [reply]
Yes, I tried that but it is downloading the outermost boundary of India. Not sure why.

**Reply by msameeruddin** [reply]
Finally, I got it after considering version 4.0.4. Thanks all.

**Comment by simhareddy677** [comment]
How to submit the taks.. could you please explain it in detail..

**Reply by msameeruddin** [reply]
![image](https://user-images.githubusercontent.com/63333753/157650750-6136867a-59b9-42a6-b6f7-2b195d6127cf.png)

**Reply by maby200** [reply]
If you get stuck with those instructions you can read the link below from that text:
[How to contribute to a GitHub Project](https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/)

**Comment by SHRAADHA** [comment]
We work on windows or linux?
software requirement shows that  we used 18.04 ubuntu .If i use anaconda for python in window 11 any problem

**Reply by Swetha-Kolluri** [reply]
No problem, you can use Python in Windows or Linux.

**Reply by SHRAADHA** [reply]
Please anyone confirm my code is send or not

**Comment by YaodongQiu** [comment]
Do we actually need to download additional data for the  task 1? or the data inside dataset is sufficient for task 1? The reason i am asking is that I can't find any relevant information about administrative boundary  of Telangana except longitude and latitude which is not usable without the proper label. 
Appreciated the help!

**Reply by Swetha-Kolluri** [reply]
Datasets are also available here in this folder:
https://github.com/undpindia/Data4Policy/tree/main/Geospatial%20Data%20Science%20Internship/Datasets

**Reply by YaodongQiu** [reply]
I think that is level 2 data which doesn't contain district information. we need level 3 so should I download level 3 from the website?

**Reply by Swetha-Kolluri** [reply]
Yes please use Level 3 in the GADM data as mentioned in the task description.

**Reply by YaodongQiu** [reply]
ok, thanks for answering my question.

**Comment by YaodongQiu** [comment]
For Task 2:  I understand we need to use the fire dataset to predict the probrability of agricultural related fire (type=0) given all other features.  

But I don't know how the ESRI LULC map can be used to propose the mythology. 

Any help apprieciated!

**Reply by msameeruddin** [reply]
But in the readme file, it is said that it is an unsupervised model (target variable will not be present). You are saying that we need to predict the probability of `type` which clearly indicates a target variable. In that case, do we need to use classification models?

**Reply by YaodongQiu** [reply]
I think you are right about that.  In fact I am also confusing about the statement that we need to predict the probability and also use unsupervised model.

**Reply by parvathykrishnank** [reply]
I have updated Task 2 description to clarify this question about the probability. Please check that.

**Reply by YaodongQiu** [reply]
Thanks for the clarification

**Comment by Vivek0000007** [comment]
I am still not clear about the data. Can anyone help?

**Reply by YaodongQiu** [reply]
for my understanding, You need to go to the website. find the level 3 data , then open and load it to the notebook. Then try to use it and the dataset from the forked repo to complete task 1.

**Reply by Vivek0000007** [reply]
Can you help in downloading?

**Reply by parvathykrishnank** [reply]
The GADM data is available here - https://gadm.org/download_country.html

<img width="923" alt="image" src="https://user-images.githubusercontent.com/42402451/157852769-328e8433-7f2a-4ccb-8b76-7338d167da8d.png">

You can download it here and extract the boundaries for Telangana

**Reply by msameeruddin** [reply]
The only problem here is we can extract the boundaries, but there are no adm names with which we can do spatial join. The geometry column has polygons but the labels are None.

**Comment by SUNITHAMC** [comment]
Hi,
I have these two questions.
1. For the Task 1 do we need to consider all types of fires or the fire due to crop burning only?
2. Do we need to consider both Aqua fire and Land fire or only the Terra/Land fire?

**Reply by parvathykrishnank** [reply]
For Task 1, please use all the fires as available in the dataset.

**Comment by parvathykrishnank** [comment, marked as answer]
Since we are receiving a few questions regarding Task 2, please find some clarifications below

**While Task 1 is a qualifying task, in this task we want to evaluate your methodology and approach and not focus solely on the results.**

Currently, we have the following dataset, which includes all the fire information as downloaded from the NASA Fire Information dataset for Telangana. 

<img width="1070" alt="image" src="https://user-images.githubusercontent.com/42402451/157853882-1bf3b54b-d1ea-49c7-b89a-e5bcfc4968df.png">

 As this is an unsupervised model, we will not specifically be testing your code for evaluation metrics but on your **approach and implementation.** The goal of this is to help us identify how you approach the problem, your research/ideas on how this can be implemented. We hope you highlight this in the Medium Blog that you will be submitting along with the package. Please make your submission even if you do not have a classification model, but only ideas on how to implement this.

**Comment by Vivek0000007** [comment]
Can anyone help with installation of geopandas? I have tried everything to install it but still unable to

**Reply by parvathykrishnank** [reply]
Maybe this helps - https://towardsdatascience.com/geopandas-installation-the-easy-way-for-windows-31a666b3610f
(assuming you are using windows).

**Reply by Vivek0000007** [reply]
These files aren't downloading

**Reply by Vivek0000007** [reply]
I tried this earlier

**Reply by msameeruddin** [reply]
https://stackoverflow.com/a/58943939

I tried the above. `pipwin` can help you. I do not use anaconda.

**Reply by Vivek0000007** [reply]
I don't know why but geopandas isn't getting installed on my laptop. I have tried everywhere and done everything. Please help

**Reply by YaodongQiu** [reply]
For me conda cause error too. to  solve the problem I did the following: 
1.use pip to install the package on me local computer not the inside conda virtual environment. 
2. select the correct interpreter on VS code.

**Comment by riazsheriff17** [comment]
Can I use R programming tool for coding, mostly I am familiar with using R?

**Reply by parvathykrishnank** [reply]
No. All the analytics have to be in Python.

**Comment by Vivek0000007** [comment]
I am not able to install geopandas, do we have any other solution?

**Reply by msameeruddin** [reply]
The only option left now is to try on Google Colab. You can easily install geopandas there.

**Comment by maby200** [comment]
Do literature references count in the 600 max words for the Medium post?

**Comment by sonof777** [comment]
The instructions for submission are a bit confusing. How do I get to gitbash on this github site to clone locally? Also, how do I create a new folder in the Geospatial Data Sciences Internship folder? I'm not too familiar with this site.

**Reply by maby200** [reply]
You can see this link which is step by step.
https://www.google.com/amp/s/www.dataschool.io/how-to-contribute-on-github/amp/

But make sure you are following the steps given from the UNDP as well:
https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/

**Reply by sonof777** [reply]
The link is a bit confusing. I can't really make sense out of it. Are there any clearer directions or is there an alternative means of submission? This is the first time I am using github. The steps I need cleared up are: forking the project Data4Policy & clone locally, creating an upstream remote and syncing my local copy, and creating a folder  inside the folder Geospatial Data Science Internship.

**Comment by Vivek0000007** [comment]
I have submitted, but I am not sure if it's in the right place. I am not familiar with the site. So please confirm.

**Reply by parvathykrishnank** [reply]
I have received your pull request

**Comment by u12ce045** [comment]
I completed the project and forked/pulled github repository and its looking in my profile. so anything remaining or there any additional step to followed.

**Reply by parvathykrishnank** [reply]
We have received your submission. Thank you!

**Comment by parvathykrishnank** [comment]
Please use the guidelines in the following link for submitting the assignments. Until the pull request is received by us the submission will not be evaluated. 

https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/

**Reply by Vivek0000007** [reply]
Can you confirm whose you have received?

**Reply by msameeruddin** [reply]
Yes that would be great.

**Reply by parvathykrishnank** [reply]
I have received both your pull request.

**Reply by Vivek0000007** [reply]
Okay thank you

**Reply by u12ce045** [reply]
can you check mine also!

**Comment by riazsheriff17** [comment]
what is the deadline time (IST)?

**Reply by msameeruddin** [reply]
I think they have specified it on the mail.

**Comment by parvathykrishnank** [comment]
Hello everyone,

The Pull Request has to be submitted before the DEADLINE - March 13th SUNDAY (11:59 PM EST - March 14th MONDAY 09:29 AM IST)

**Reply by SHRAADHA** [reply]
Mam , shradha can you check code is submitted from my side.

**Reply by SHRAADHA** [reply]
file name is undp-assesment-test

**Reply by parvathykrishnank** [reply]
Yes, we have received your pull request. Thank you.

**Comment by riazsheriff17** [comment]
I have submitted my assignment as Riaz sheriff task, pls check whether it has uploaded.

**Reply by parvathykrishnank** [reply]
Yes, we have received your pull request. Thank you.

**Comment by Anjali1808** [comment]
Please check if my assignment is uploaded.

**Reply by parvathykrishnank** [reply]
@Anjali1808 - Hello, I have not yet received your Pull Request. Please make sure you have followed all the steps outlined in the Submission Process. Thank you.

**Reply by Anjali1808** [reply]
Thank you for the information. Can you please check once again?

**Reply by parvathykrishnank** [reply]
Yes, we have received your pull request. Thank you.

**Comment by YaodongQiu** [comment]
Hello dear Mantainer:

I have submitted pull request for my task as Yaodong Qiu

Thank you for this project and I feel great for the learning experience

**Reply by parvathykrishnank** [reply]
Hello, 

We have received your pull request. Thank you for the submission.

**Comment by sonof777** [comment]
I tried to work it out. I'm not sure if my submission has been sent.

**Comment by yulinhong99** [comment]
Hi all!
Because of my laptop's probelm, I used my friends laptop to uploaded the submission foler, but we don't know why it still shows my friends name. We will try to figure out.(I did not submitted late). Thank you so much!

**Reply by yulinhong99** [reply]
![1647231056(1)](https://user-images.githubusercontent.com/101537168/158103661-debe321e-9efa-4d41-86dc-1e441dfe269e.png)
I uploaded my solution at 11:40 pm New York time, However, I still cannot change the name, So I keep uploading it. But it failed. Sorry for the inconvenience, but the folder's name is correct.

**Reply by sonof777** [reply]
I sent the pull request and uploaded my solution yesterday. Did it come up? I didn't get any response after I sent my query, so, I just wanted to make sure.

**Comment by u12ce045** [comment]
When will be the result declared, because project will be started in 21st March?

**Comment by riazsheriff17** [comment]
is the result declared ?

**Comment by Swetha-Kolluri** [comment]
Dear All,
Thank you for your submissions. We have received some great ideas from you all. We have already invited some of you for the next round of interviews based on:
(1) Current enrollment in a university (eligibility) 
(2) Completion of both the tasks in written/technical round 
(3) Quality of code & blog
The final result will be announced very soon.
Meanwhile keep exploring [UNDP Careers](https://www.undp.org/careers) for future opportunities 
Goodluck everyone!

**Reply by Vivek0000007** [reply]
Can you send the shortlist here?

**Reply by Swetha-Kolluri** [reply]
Sorry that's not possible. All official communication will be over email. If you are shorlisted, you would have already received the email (pls check the spam folder as well).
